packages-next/admin-ui/src/pages/ListPage/index.tsx

```diff
@@ -77,11 +77,23 @@ let listMetaGraphqlQuery: TypedDocumentNode<
 function useQueryParamsFromLocalStorage(listKey: string) {
   const router = useRouter();
   const localStorageKey = `keystone.list.${listKey}.list.page.info`;
+  const cacheableQueries = ['sortBy', 'fields'];
+
+  const resetToDefaults = () => {
+    localStorage.removeItem(`keystone.list.${listKey}.list.page.info`);
+    router.replace({
+      query: null,
+    });
+  };
+
+  // GET QUERY FROM CACHE IF CONDITIONS ARE RIGHT
+  // MERGE QUERY PARAMS FROM CACHE WITH QUERY PARAMS FROM ROUTER
   useEffect(() => {
-    let hasSomeQueryParamsWhichAreAboutListPage = Object.keys(router.query).some(
-      x => x.startsWith('!') || x === 'page' || x === 'pageSize' || x === 'fields'
-    );
-    if (!hasSomeQueryParamsWhichAreAboutListPage) {
+    let hasSomeQueryParamsWhichAreAboutListPage = Object.keys(router.query).some(x => {
+      return x.startsWith('!') || cacheableQueries.includes(x);
+    });
+
+    if (!hasSomeQueryParamsWhichAreAboutListPage && router.isReady) {
```

Comment:
What does router.isReady do here?

Reply:
as per the docs https://nextjs.org/docs/api-reference/next/router#router-object
it lets us know when the router fields are updated client side. 

The bug we were experience was due to the fact that the !hasSomeQueryParams... check was being run on an empty query object, despite them being present on the path when we run the effect. Most likely due to the fact that the check was happening before the router object had been updated